Integer overflow panic in fcntl F_SETLK/F_SETLKW

### Describe the bug

The bug occurs in the fcntl system call handler when processing file locking operations (F_SETLK/F_SETLKW). When the l_start and l_len fields of the struct flock contain large values (e.g., INT64_MAX, UINT64_MAX), an arithmetic addition operation overflows, triggering a Rust panic: "attempt to add with overflow".

https://github.com/asterinas/asterinas/blob/82ccfcd4c6f69d538e3fd62153d7245b7d2fcaa1/kernel/src/syscall/fcntl.rs#L284

### To Reproduce

Compile the program and run:

```
#define _GNU_SOURCE
#include <unistd.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <limits.h>

int main(void) {
    char template[] = "/tmp/repro_fcntl_XXXXXX";
    int fd = mkstemp(template);
    if (fd < 0) {
        perror("mkstemp");
        return 1;
    }
    unlink(template);

    struct flock fl;
    fl.l_type = F_WRLCK;
    fl.l_whence = SEEK_SET;
    // start + len will overflow if start is negative and len is INT64_MIN
    fl.l_start = -1; 
    fl.l_len = INT64_MIN; 

    printf("Attempting fcntl with l_start=%ld, l_len=%ld\n", fl.l_start, fl.l_len);
    
    // This should trigger the overflow at kernel/src/syscall/fcntl.rs:284
    // let new_start = start + len;
    if (fcntl(fd, F_SETLK, &fl) == -1) {
        perror("fcntl");
    }

    close(fd);
    return 0;
}

```

### Environment

- Official docker asterinas/asterinas:0.17.0-20251217
- Asterinas version: main  [#c91d383c91a5dccb057fa7f315657ed73ff8e0e6](https://github.com/asterinas/asterinas/commit/c91d383c91a5dccb057fa7f315657ed73ff8e0e6)

### Logs

```
# ./fuzzer_arithmetic_overflow

[   114.623] ERROR: Uncaught panic:
        attempt to add with overflow
        at /root/asterinas/kernel/src/syscall/fcntl.rs:284
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, is_exited: false, cpu_affinity: AtomicIdSet { bits: [1], phantom: PhantomData<ostd::cpu::id::CpuId> }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31493420 }, fair: FairAttr { weight: 1024, pending_weight: 0, vruntime: 346639386 } } })
Printing stack trace:
   1: fn 0xffffffff88a0e290 - pc 0xffffffff88a0e2db / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70430;

   2: fn 0xffffffff886e7740 - pc 0xffffffff886e7f3d / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70450;

   3: fn 0xffffffff886e95b0 - pc 0xffffffff886e95ba / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70820;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70830;

   5: fn 0xffffffff88af8560 - pc 0xffffffff88af85a6 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70840;

   6: fn 0xffffffff88b02880 - pc 0xffffffff88b028a5 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70880;

   7: fn 0xffffffff881388c0 - pc 0xffffffff88138e89 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70890;

   8: fn 0xffffffff88137190 - pc 0xffffffff88137627 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70c00;

   9: fn 0xffffffff88138fa0 - pc 0xffffffff88139338 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd70fb0;

  10: fn 0xffffffff883005a0 - pc 0xffffffff88311c1e / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd710f0;

  11: fn 0xffffffff882934b0 - pc 0xffffffff8829354a / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd844f0;

  12: fn 0xffffffff884c1050 - pc 0xffffffff884c1901 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84650;

  13: fn 0xffffffff884c1ac0 - pc 0xffffffff884c1b1e / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84b40;

  14: fn 0xffffffff884a0af0 - pc 0xffffffff884a0b1e / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84cf0;

  15: fn 0xffffffff884a0a20 - pc 0xffffffff884a0a66 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84d20;

  16: fn 0xffffffff881ce770 - pc 0xffffffff881ce79c / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84d90;

  17: fn 0xffffffff88144f70 - pc 0xffffffff88144faf / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84dd0;

  18: fn 0xffffffff884c1a80 - pc 0xffffffff884c1aa7 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84f40;

  19: fn 0xffffffff880e4910 - pc 0xffffffff880e492d / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84f80;

  20: fn 0xffffffff889a8150 - pc 0xffffffff889a8175 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84fa0;

  21: fn 0xffffffff88961760 - pc 0xffffffff889617e4 / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd84fd0;

  22: fn 0xffffffff88a166b5 - pc 0xffffffff88a166be / registers:

     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd85000;


     rax 0xffffdfffffd70348; rdx 0xffffffff889609a2; rcx  0x10000020100e82d; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88beb1b9; rbp                0x0; rsp 0xffffdfffffd85008;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) ostd::panic::print_stack_trace
          at /root/asterinas/ostd/src/panic.rs:113
(  2) aster_nix::thread::oops::panic_handler
          at /root/asterinas/kernel/src/thread/oops.rs:130
(  3) __ostd_panic_handler
          at ??:?
(  4) __rustc::rust_begin_unwind
          at 4xmqay6d0res9hyu5b0rjoyzh:?
(  5) core::panicking::panic_fmt
          at ??:?
(  6) core::panicking::panic_const::panic_const_add_overflow
          at ??:?
(  7) aster_nix::syscall::fcntl::from_c_flock_and_file
          at 23fw1rxa2g8lfbo3k3fsrtjek:?
(  8) aster_nix::syscall::fcntl::handle_setlk
          at /root/asterinas/kernel/src/syscall/fcntl.rs:123
(  9) aster_nix::syscall::fcntl::sys_fcntl
          at /root/asterinas/kernel/src/syscall/fcntl.rs:28
( 10) aster_nix::syscall::arch::syscall_dispatch
          at /root/asterinas/kernel/src/syscall/mod.rs:217
( 11) aster_nix::syscall::handle_syscall
          at /root/asterinas/kernel/src/syscall/mod.rs:374
( 12) aster_nix::thread::task::create_new_user_task::{closure#0}
          at /root/asterinas/kernel/src/thread/task.rs:95
( 13) aster_nix::thread::task::create_new_user_task::{closure#1}
          at /root/asterinas/kernel/src/thread/task.rs:126
( 14) unwinding::panicking::catch_unwind::do_call::<aster_nix::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:56
( 15) unwinding::panicking::catch_unwind::<unwinding::panic::RustPanic, (), aster_nix::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:42
( 16) unwinding::panic::catch_unwind::<(), aster_nix::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panic.rs:87
( 17) aster_nix::thread::oops::catch_panics_as_oops::<aster_nix::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/asterinas/kernel/src/thread/oops.rs:55
( 18) aster_nix::thread::task::create_new_user_task::{closure#2}
          at /root/asterinas/kernel/src/thread/task.rs:131
( 19) <aster_nix::thread::task::create_new_user_task::{closure#2} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 20) <alloc::boxed::Box<dyn core::ops::function::FnOnce<(), Output = ()> + core::marker::Send> as core::ops::function::FnOnce<()>>::call_once
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2206
( 21) kernel_task_entry
          at /root/asterinas/ostd/src/task/mod.rs:196
( 22) kernel_task_entry_wrapper
          at ??:?
make: *** [Makefile:308: run_kernel] Error 1
```

--- a/kernel/src/syscall/fcntl.rs
+++ b/kernel/src/syscall/fcntl.rs
@@ -271,6 +271,10 @@ fn from_c_flock_and_file(lock: &c_flock, file: &dyn FileLike) -> Result<FileRang
         }
     };
 
+    if start < 0 {
+        return Err(Error::with_message(Errno::EINVAL, "invalid start"));
+    }
+
     let (start, end) = match lock.l_len {
         len if len > 0 => {
             let end = start
@@ -281,6 +285,7 @@ fn from_c_flock_and_file(lock: &c_flock, file: &dyn FileLike) -> Result<FileRang
         0 => (start as usize, OFFSET_MAX),
         len if len < 0 => {
             let end = start;
+            // `start + len` won't overflow because `start >= 0` and `len < 0`.
             let new_start = start + len;
             if new_start < 0 {
                 return Err(Error::with_message(Errno::EINVAL, "invalid len"));

